x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -269,13 +268,10 @@ public void validate(ByteSizeValue value, Map<Setting<?>, Object> settings, bool
 
     private final int numRegions;
     private final ConcurrentLinkedQueue<SharedBytes.IO> freeRegions = new ConcurrentLinkedQueue<>();
```

Comment:
Should we push the free-list into the cache interface as well maybe? Then we could do away with the `evict` call and instead do `evictAndGet` on the cache directly?